theseus/embodied/kinematics/robot_model.py

```diff
@@ -40,3 +41,34 @@ def forward_kinematics(self, robot_pose: RobotModelInput) -> Dict[str, LieGroup]
 
     def dim(self) -> int:
         return 1
+
+
+class UrdfRobotModel(KinematicsModel):
+    def __init__(self, urdf_path: str):
+        self.drm_model = drm.DifferentiableRobotModel(urdf_path)
+
+    def forward_kinematics(self, joint_states: RobotModelInput) -> Dict[str, LieGroup]:
+        """Computes forward kinematics
+        Args:
+            joint_states: Vector of all joint angles
+        Outputs:
+             Dictionary that maps link name to link pose
+
+        (Note: differentiable robot model uses the xyzw quaternion convention,
+        while theseus uses wxyz, hence the conversion.)
+        """
+        link_poses: Dict[str, LieGroup] = {}
+        for link_name in self.drm_model.get_link_names():
```

Comment:
Not sure if this needs to happen on DRM side but the intermediate manipulator jacobians can be reused/cached for efficiency when many/all links are being evaluated (does this already happen under the hood?).

Reply:
I agree that this is quite inefficient for now especially for models with a lot of links, as this is the only API for forward kinematics DRM provides. I will prefer to refactor this on the DRM side, and create another PR in Theseus to adapt.
Let me know if there are any timeline dependencies for this task!

Reply:
Added issue #123 to track this.